### DISCUSSION TITLE
What happened with the --commits-file argument?

### DISCUSSION THREAD

**Discussion by syphernl** [zricethezav/gitleaks#727] [Q&A] [answered]
We've been using Gitleaks for a while now. In order to speed things up we were providing a list of commits using the `--commits-file` argument, which was part of the documentation IIRC.
However, since the release of Gitleaks v8 this seems to have vanished into thin air without any mention of this argument in the release notes or in the [announcement post](https://github.com/zricethezav/gitleaks/issues/683).

We use this check within PR checks which retrieves all commits between `develop` and the current branch using:
```
git log --left-right --cherry-pick --pretty=format:"%H" refs/remotes/origin/$CI_DEFAULT_BRANCH...refs/remotes/origin/$CI_COMMIT_REF_NAME > "$CI_COMMIT_SHA"_commit_list.txt
```

This file was then fed into gitleaks using `--commits-file=${GITLEAKS_COMMITS_FILE}`.

What would be the v8 equivalent of doing this?

**Comment by zricethezav** [comment, marked as answer]
@syphernl 
> However, since the release of Gitleaks v8 this seems to have vanished into thin air without any mention of this argument in the release notes or in the announcement post.

```
Removed all commit options
    All commit options can be supported by the --log-opts argument. --log-opts accepts any git log options.
```

You can use `--log-opts` and use the first and last commit from the commits file, so something like `--log-opts=commitA..commitB`. Or you can come up with a clever way how to get the first and last commit
